### PR TITLE
Dbz-6484 Added logic to support table name when its passed in this format dbname.tablename

### PR DESCRIPTION
**Pull request by subkanthi**
- Changed JDBC password to optional.
- Added function to split database and table name.

### CODE REVIEW

File: debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/JdbcCommonConfig.java [hunk 1]
```diff
@@ -37,7 +37,7 @@ public class JdbcCommonConfig {
 
     public static final Field PROP_PASSWORD = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "password")
             .withDescription("Password of the database which will be used to access the database storage")
-            .withValidation(Field::isRequired);
+            .withValidation(Field::isOptional);
```
Comment (jpechane) [inline debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/JdbcCommonConfig.java:40]: Why this change? It seems unrelated.

File: debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistory.java [hunk 2]
```diff
@@ -184,7 +184,9 @@ public boolean storageExists() {
         boolean sExists = false;
         try {
             DatabaseMetaData dbMeta = conn.getMetaData();
-            ResultSet tableExists = dbMeta.getTables(null, null, config.getTableName(), null);
+            String databaseName = config.getDatabaseName();
+            ResultSet tableExists = dbMeta.getTables(databaseName != null ? databaseName : null,
```
Comment (jpechane) [inline debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistory.java:188]: The `?:` oprator is superfluous, if `databaseName` is `null` then it can be used instead of constant.

**Review by jpechane — CHANGES_REQUESTED** [review]
@subkanthi Thanks for the PR. I left two comments. Could you please also drop the merge commit for the PR and keep only change commits? Thanks

File: debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/JdbcCommonConfig.java (hunk 1 above)
Comment (subkanthi) [reply debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/JdbcCommonConfig.java:40]: Hi @jpechane , sorry its not related, its just that there might be cases where you might not have to provide a password, especially connecting to local , dev databases. 
Should I remove it?

File: debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistory.java [hunk 3]
```diff
@@ -116,7 +116,7 @@ protected void storeRecord(HistoryRecord record) throws SchemaHistoryException {
                     sql.setInt(3, partSeq);
                     sql.setTimestamp(4, currentTs);
                     sql.setInt(5, recordInsertSeq.incrementAndGet());
-                    sql.executeUpdate();
+                    sql.execute();
```
Comment (vjuranek) [inline debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistory.java:119]: Why is this change needed?

File: debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistory.java [hunk 4]
```diff
@@ -126,7 +126,7 @@ protected void storeRecord(HistoryRecord record) throws SchemaHistoryException {
                     conn.rollback();
                 }
                 catch (SQLException ex) {
-                    // ignore
+                    LOG.error("Exception during storeRecord", ex);
```
Comment (vjuranek) [inline debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistory.java:129]: We throw the exception on the next line, no need to log anything here, it just duplicates the exception in the log.

File: debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistoryConfig.java [hunk 5]
```diff
@@ -22,7 +22,7 @@
 public class JdbcSchemaHistoryConfig extends JdbcCommonConfig {
 
     private static final String DEFAULT_TABLE_NAME = "debezium_database_history";
-    private static final Field PROP_TABLE_NAME = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "schema.history.table.name")
+    public static final Field PROP_TABLE_NAME = Field.create(CONFIGURATION_FIELD_PREFIX_STRING + "schema.history.table.name")
```
Comment (vjuranek) [inline debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistoryConfig.java:25]: do we need to make it public?

**Review by vjuranek — CHANGES_REQUESTED** [review]
There are some smaller issue (unneeded/unrelated changes), but overall LGTM. Also please a test for it. I guess just a unit test in `JdbcSchemaHistoryTest` would be sufficient. Thanks!

File: debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistoryConfig.java (hunk 5 above)
Comment (subkanthi) [reply debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistoryConfig.java:25]: Hi @vjuranek , when we use debezium engine and if we need to get the value from properties, it might be better to use the field, the other ones are `public` for the same reason.

File: debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistory.java (hunk 4 above)
Comment (subkanthi) [reply debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistory.java:129]: reverted.

File: debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistory.java (hunk 3 above)
Comment (subkanthi) [reply debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistory.java:119]: reverted change.

Comment (subkanthi) [reply debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistory.java:119]: will cover this in another PR for mysql.

**Comment by subkanthi** [comment]
> There are some smaller issue (unneeded/unrelated changes), but overall LGTM. Also please a test for it. I guess just a unit test in `JdbcSchemaHistoryTest` would be sufficient. Thanks!

Hi @vjuranek , please take a look whenever u get a chance.

File: debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistoryConfig.java (hunk 5 above)
Comment (vjuranek) [reply debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistoryConfig.java:25]: I'd make it public when we need make it public

**Review by vjuranek — CHANGES_REQUESTED** [review]
Overall LGMT, but please revert unrelated changes. Thanks!

File: debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistoryConfig.java (hunk 5 above)
Comment (vjuranek) [reply debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/history/JdbcSchemaHistoryConfig.java:25]: ops, sorry, I missed this one

**Review by vjuranek — APPROVED** [review]

**Comment by vjuranek** [comment]
Thanks @subkanthi , LGTM now. Let's wait for @jpechane if he has any further comments.

**Comment by subkanthi** [comment]
> Thanks @subkanthi , LGTM now. Let's wait for @jpechane if he has any further comments.

Hi @jpechane , can you please take a look when you get a chance

**Review by jpechane — APPROVED** [review]
LGTM, thanks!
@subkanthi Could you please rebase it to the latest `main` as there are conflicts?

@vjuranek Feel free to merge the PR as soon as it is possible to do it.

**Comment by subkanthi** [comment]
> LGTM, thanks! @subkanthi Could you please rebase it to the latest `main` as there are conflicts?
> 
> @vjuranek Feel free to merge the PR as soon as it is possible to do it.

Thanks, rebased with lates.

**Comment by vjuranek** [comment]
hi @subkanthi , GH says there are still conflict. Could you please rebase once again? Thanks!

**Comment by vjuranek** [comment]
hi @subkanthi , unfortunately what you did is not a rebase but a merge and there are lots of conflict. Could you please do the rebase so that your commit can be applied cleanly? Thanks!

**Comment by vjuranek** [comment]
@subkanthi I could do it myself if there are separate commits, but the main issue is [this commit](https://github.com/ismailsimsek/debezium/commit/3781cf3eca3854f93fe601cff0f5a06b69132140) where you merge everything with other changes on which your commit should be rebased. With this is impossible for to get the required commits as they don't exists.

**Comment by subkanthi** [comment]
> @subkanthi I could do it myself if there are separate commits, but the main issue is [this commit](https://github.com/ismailsimsek/debezium/commit/3781cf3eca3854f93fe601cff0f5a06b69132140) where you merge everything with other changes on which your commit should be rebased. With this is impossible for to get the required commits as they don't exists.

Hi @vjuranek , I see a lot of conflicts when I do a rebase, should I just create a new PR with the changes
```
git pull --rebase origin main
From github.com:ismailsimsek/debezium
 * branch                main       -> FETCH_HEAD
Auto-merging .github/workflows/debezium-workflow.yml
CONFLICT (content): Merge conflict in .github/workflows/debezium-workflow.yml
Auto-merging CHANGELOG.md
CONFLICT (content): Merge conflict in CHANGELOG.md
Auto-merging COPYRIGHT.txt
Auto-merging debezium-bom/pom.xml
Auto-merging debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ChangeStreamPipelineFactory.java
Auto-merging debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorConfig.java
CONFLICT (content): Merge conflict in debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbConnectorConfig.java
Auto-merging debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSnapshotChangeEventSource.java
CONFLICT (content): Merge conflict in debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbSnapshotChangeEventSource.java
Auto-merging debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/MongoDbStreamingChangeEventSource.java
Auto-merging debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ReplicaSetOffsetContext.java
CONFLICT (content): Merge conflict in debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/ReplicaSetOffsetContext.java
Auto-merging debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/connection/ConnectionContext.java
CONFLICT (content): Merge conflict in debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/connection/ConnectionContext.java
Auto-merging debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/connection/MongoDbConnection.java
CONFLICT (content): Merge conflict in debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/connection/MongoDbConnection.java
CONFLICT (modify/delete): debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/AbstractBaseMongoIT.java deleted in HEAD and modified in 3781cf3ec (DBZ-6484 Added logic to split database name and table name for JDBC offset storage).  Version 3781cf3ec (DBZ-6484 Added logic to split database name and table name for JDBC offset storage) of debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/AbstractBaseMongoIT.java left in tree.
Auto-merging debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/AbstractMongoIT.java
CONFLICT (content): Merge conflict in debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/AbstractMongoIT.java
Auto-merging debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/junit/MongoDbDatabaseProvider.java
CONFLICT (content): Merge conflict in debezium-connector-mongodb/src/test/java/io/debezium/connector/mongodb/junit/MongoDbDatabaseProvider.java
Auto-merging debezium-connector-mysql/pom.xml
Auto-merging debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlSnapshotChangeEventSource.java
Auto-merging debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/MySqlParserIT.java
Auto-merging debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/SnapshotParallelSourceIT.java
CONFLICT (content): Merge conflict in debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/SnapshotParallelSourceIT.java
Auto-merging debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/SnapshotSourceIT.java
CONFLICT (content): Merge conflict in debezium-connector-mysql/src/test/java/io/debezium/connector/mysql/SnapshotSourceIT.java
Auto-merging debezium-connector-oracle/pom.xml
Auto-merging debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnection.java
Auto-merging debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java
CONFLICT (content): Merge conflict in debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleConnectorConfig.java
Auto-merging debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/OracleValueConverters.java
Auto-merging debezium-connector-oracle/src/main/java/io/debezium/connector/oracle/logminer/LogMinerStreamingChangeEventSource.java
Auto-merging debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/OracleConnectorIT.java
CONFLICT (content): Merge conflict in debezium-connector-oracle/src/test/java/io/debezium/connector/oracle/OracleConnectorIT.java
Auto-merging debezium-connector-postgres/pom.xml
Auto-merging debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/PostgresValueConverter.java
Auto-merging debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/AbstractRecordsProducerTest.java
Auto-merging debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/RecordsStreamProducerIT.java
Auto-merging debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/spi/CustomActionProvider.java
CONFLICT (add/add): Merge conflict in debezium-connector-postgres/src/test/java/io/debezium/connector/postgresql/spi/CustomActionProvider.java
Auto-merging debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnectorConfig.java
CONFLICT (content): Merge conflict in debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnectorConfig.java
Auto-merging debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java
CONFLICT (content): Merge conflict in debezium-core/src/main/java/io/debezium/config/CommonConnectorConfig.java
Auto-merging debezium-core/src/main/java/io/debezium/config/Configuration.java
Auto-merging debezium-core/src/main/java/io/debezium/metrics/Metrics.java
Auto-merging debezium-core/src/main/java/io/debezium/pipeline/JmxUtils.java
CONFLICT (add/add): Merge conflict in debezium-core/src/main/java/io/debezium/pipeline/JmxUtils.java
Auto-merging debezium-core/src/main/java/io/debezium/pipeline/notification/IncrementalSnapshotNotificationService.java
CONFLICT (add/add): Merge conflict in debezium-core/src/main/java/io/debezium/pipeline/notification/IncrementalSnapshotNotificationService.java
Auto-merging debezium-core/src/main/java/io/debezium/relational/RelationalSnapshotChangeEventSource.java
Auto-merging debezium-core/src/test/java/io/debezium/junit/logging/LogInterceptor.java
CONFLICT (content): Merge conflict in debezium-core/src/test/java/io/debezium/junit/logging/LogInterceptor.java
Auto-merging debezium-core/src/test/java/io/debezium/pipeline/notification/IncrementalSnapshotNotificationServiceTest.java
CONFLICT (add/add): Merge conflict in debezium-core/src/test/java/io/debezium/pipeline/notification/IncrementalSnapshotNotificationServiceTest.java
Auto-merging debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4
CONFLICT (content): Merge conflict in debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlLexer.g4
Auto-merging debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4
CONFLICT (content): Merge conflict in debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/mysql/generated/MySqlParser.g4
Auto-merging debezium-ddl-parser/src/main/antlr4/io/debezium/ddl/parser/oracle/generated/PlSqlParser.g4
Auto-merging debezium-ddl-parser/src/test/resources/mysql/examples/ddl_create.sql
Auto-merging debezium-ddl-parser/src/test/resources/oracle/examples/ddl_alter.sql
Auto-merging debezium-embedded/src/test/java/io/debezium/embedded/AbstractConnectorTest.java
CONFLICT (content): Merge conflict in debezium-embedded/src/test/java/io/debezium/embedded/AbstractConnectorTest.java
Auto-merging debezium-embedded/src/test/java/io/debezium/pipeline/notification/AbstractNotificationsIT.java
CONFLICT (content): Merge conflict in debezium-embedded/src/test/java/io/debezium/pipeline/notification/AbstractNotificationsIT.java
Auto-merging debezium-quarkus-outbox-common/deployment/pom.xml
Auto-merging debezium-storage/debezium-storage-jdbc/src/main/java/io/debezium/storage/jdbc/offset/JdbcOffsetBackingStore.java
Auto-merging debezium-testing/debezium-testing-system/pom.xml
CONFLICT (content): Merge conflict in debezium-testing/debezium-testing-system/pom.xml
Auto-merging debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/databases/postgresql/OcpPostgreSqlReplicaController.java
CONFLICT (add/add): Merge conflict in debezium-testing/debezium-testing-system/src/main/java/io/debezium/testing/system/tools/databases/postgresql/OcpPostgreSqlReplicaController.java
Auto-merging debezium-testing/debezium-testing-testcontainers/pom.xml
Auto-merging debezium-testing/debezium-testing-testcontainers/src/main/java/io/debezium/testing/testcontainers/DebeziumContainer.java
Auto-merging documentation/antora.yml
CONFLICT (content): Merge conflict in documentation/antora.yml
Auto-merging documentation/modules/ROOT/pages/configuration/notification.adoc
CONFLICT (add/add): Merge conflict in documentation/modules/ROOT/pages/configuration/notification.adoc
Auto-merging documentation/modules/ROOT/pages/configuration/signalling.adoc
CONFLICT (content): Merge conflict in documentation/modules/ROOT/pages/configuration/signalling.adoc
Auto-merging documentation/modules/ROOT/pages/connectors/db2.adoc
Auto-merging documentation/modules/ROOT/pages/connectors/mongodb.adoc
Auto-merging documentation/modules/ROOT/pages/connectors/mysql.adoc
Auto-merging documentation/modules/ROOT/pages/connectors/oracle.adoc
Auto-merging documentation/modules/ROOT/pages/connectors/postgresql.adoc
Auto-merging documentation/modules/ROOT/pages/connectors/sqlserver.adoc
Auto-merging documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-kafka.adoc
CONFLICT (add/add): Merge conflict in documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot-kafka.adoc
Auto-merging documentation/modules/ROOT/partials/modules/all-connectors/proc-triggering-an-incremental-snapshot.adoc
Auto-merging jenkins-jobs/pipelines/release/release-pipeline.groovy
Auto-merging jenkins-jobs/scripts/config/Aliases.txt
CONFLICT (content): Merge conflict in jenkins-jobs/scripts/config/Aliases.txt
Auto-merging pom.xml
CONFLICT (content): Merge conflict in pom.xml
error: could not apply 3781cf3ec... DBZ-6484 Added logic to split database name and table name for JDBC offset storage
hint: Resolve all conflicts manually, mark them as resolved with
hint: "git add/rm <conflicted_files>", then run "git rebase --continue".
hint: You can instead skip this commit: run "git rebase --skip".
hint: To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 3781cf3ec... DBZ-6484 Added logic to split database name and table name for JDBC offset storage
```

**Comment by vjuranek** [comment]
@subkanthi yes, this is what I mentioned above. No need to create new PR, just extract the changes, rebase them on current main branch and force push to [ismailsimsek:dbz-6484](https://github.com/ismailsimsek/debezium/tree/dbz-6484)

**Comment by subkanthi** [comment]
> @subkanthi yes, this is what I mentioned above. No need to create new PR, just extract the changes, rebase them on current main branch and force push to [ismailsimsek:dbz-6484](https://github.com/ismailsimsek/debezium/tree/dbz-6484)

@vjuranek can u please check now.

**Comment by jpechane** [comment]
@subkanthi Unfortunately, this PR ended in very bad shape given all the merges in the history. I created a new one (https://github.com/debezium/debezium/pull/4739) that contains just the net changes. If tests pass we can merge it.
I am sorry for that but it was the most effective solution right now.
